include/zephyr/device.h

```diff
@@ -980,12 +995,25 @@ static inline bool z_impl_device_is_ready(const struct device *dev)
 		Z_INIT_ENTRY_NAME(DEVICE_NAME_GET(dev_id)) = {                                     \
 			.init_fn = {COND_CODE_1(Z_DEVICE_IS_MUTABLE(node_id), (.dev_rw), (.dev)) = \
 					    (init_fn_)},                                           \
-			{                                                                          \
-				COND_CODE_1(Z_DEVICE_IS_MUTABLE(node_id), (.dev_rw), (.dev)) =     \
-					&DEVICE_NAME_GET(dev_id),                                  \
-			},                                                                         \
+			Z_DEVICE_INIT_ENTRY_DEV(node_id, dev_id),                                  \
 	}
 
+/*
+ * Anonymous unions require C11. Some pre-C11 gcc versions have early
```

Comment:
this is repeating information, can’t it be put somewhere and reference it?

Reply:
I tried to summarize it but I agree it's still a bit long and duplicated... any suggestion of a good place? I couldn't think of a really better way to document this sorry.

Reply:
Maybe Zephyr coding guidelines? It looks like we'll need to apply this in other places if we want C++20 compatibility. An explanation there, plus a code snippet would be a solution, you could then just say "Refer to doc/xxx.rst"

Reply:
Thanks, I'll take a look in the next couple weeks.

Reply:
Done in #74047